test/extended/networking/egressip.go

```diff
@@ -0,0 +1,654 @@
+)
+
+const (
+	// for all tests
+	namespacePrefix = "egressip"
+	egressIPYaml    = "egressip.yaml"
+	probePodName    = "prober-pod"
+
+	// for tests against host networked pods
+	egressIPTargetHostPortMin = 32667
+	egressIPTargetHostPortMax = 32767
+
+	// max time that we wait for changes to EgressIP objects
+	// to propagate to the CloudPrivateIPConfig objects
+	// this can take a significant amount of time on Azure
+	// BZ https://bugzilla.redhat.com/show_bug.cgi?id=2073045
+	egressUpdateTimeout = 180
+)
+
+var _ = g.Describe("[sig-network][Feature:EgressIP]", func() {
+	oc := exutil.NewCLI(namespacePrefix)
+	portAllocator := NewPortAllocator(egressIPTargetHostPortMin, egressIPTargetHostPortMax)
+
+	var (
+		clientset      kubernetes.Interface
+		tmpDirEgressIP string
+
+		workerNodesOrdered    []corev1.Node
+		workerNodesOrderedStr []string
+		// egressIPNodesOrdered    []corev1.Node
+		egressIPNodesOrderedStr []string
+		// nonEgressIPNode         corev1.Node
+		nonEgressIPNodeStr string
+
+		egressIPNamespace      string
+		externalNamespace      string
+		packetSnifferDaemonSet *v1.DaemonSet
+
+		ingressDomain string
+
+		cloudType configv1.PlatformType
+		hasIPv4   bool
+		hasIPv6   bool
+
+		targetProtocol string
+		targetHost     string
+		targetPort     int
+	)
+
+	InOVNKubernetesContext(func() {
+		g.BeforeEach(func() {
+			g.By("Creating a temp directory")
+			var err error
+			tmpDirEgressIP, err = ioutil.TempDir("", "egressip-e2e")
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Getting the clientset")
+			f := oc.KubeFramework()
+			clientset = f.ClientSet
+
+			g.By("Determining the cloud infrastructure type")
+			infra, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			cloudType = infra.Spec.PlatformSpec.Type
+
+			g.By("Determining the cloud address families")
+			hasIPv4, hasIPv6, err = GetIPAddressFamily(oc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Determining the target protocol, host and port")
+			targetProtocol, targetHost, targetPort, err = getTargetProtocolHostPort(oc, hasIPv4, hasIPv6, cloudType)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			framework.Logf("Testing against: Protocol %s, TargetHost: %s, TargetPort: %d", targetProtocol, targetHost, targetPort)
+
+			g.By("Creating a namespace for the prober pod")
+			// Create a target namespace and assign source and target namespace
+			// to variables for later use.
+			egressIPNamespace = f.Namespace.Name
+			externalNamespace = oc.SetupNamespace()
+
+			g.By("Getting all worker nodes in alphabetical order")
+			// Get all worker nodes, order them alphabetically with stable
+			// sort order.
+			workerNodesOrdered, err = getWorkerNodesOrdered(clientset)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			for _, s := range workerNodesOrdered {
+				workerNodesOrderedStr = append(workerNodesOrderedStr, s.Name)
```

Comment:
workerNodesOrderedNames instead of ...Str?

Reply:
Done